libs/services/designer-client-services/src/lib/standard/search.ts

```diff
@@ -125,3 +128,22 @@ function filterStateful(operation: DiscoveryOperation<BuiltInOperation> | Connec
     ? operation.properties.capabilities.includes('Stateful') || !operation.properties.capabilities.includes('Stateless')
     : operation.properties.capabilities.includes('Stateless') || !operation.properties.capabilities.includes('Stateful');
 }
+
+function filterAPIConnectors(rawConnectors: Connector[], isAzureConnectorsEnabled: boolean): Connector[] {
+  return rawConnectors.filter((rawConnector: Connector) => {
+    return !equals(rawConnector.name, 'dataOperationNew') && (!needsAzureConnection(rawConnector) || isAzureConnectorsEnabled);
+  });
+}
+
+function filterAPIOperations(
+  rawOperations: DiscoveryOperation<SomeKindOfAzureOperationDiscovery>[],
+  isAzureConnectorsEnabled: boolean
+): DiscoveryOperation<SomeKindOfAzureOperationDiscovery>[] {
+  return rawOperations.filter(
+    (rawOperation: DiscoveryOperation<SomeKindOfAzureOperationDiscovery>) => !needsAzureConnection(rawOperation) || isAzureConnectorsEnabled
+  );
+}
```

Comment:
On these two functions, if `isAzureConnectorsEnabled` is true it just returns the whole array, so we can pull that out of the filter functions (see above comment)